### PR TITLE
force email to be downcase when creating a user and loging in

### PR DESCRIPTION
**Pull request by maxvet**
Maintenant l'adresse courriel sera en minuscule lors de la création d'un utilisateur et lorsqu'un utilisateur se connectera à son compte.

### CODE REVIEW

File: app/graphql/mutations/auth/sign_in.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ class Mutations::Auth::SignIn < GraphQL::Schema::Mutation
 
   def resolve(email:, password:, remember_me:)
     response = context[:response]
-
+    email = email.downcase
     user = User.find_by email: email
```
Comment (ferland182) [inline app/graphql/mutations/auth/sign_in.rb]: The `email.downcase` can be used like this: `User.find_by email: email.downcase` as the email variable will not be use afterwards

File: app/graphql/types/auth/user.rb [hunk 2]
```diff
@@ -10,4 +10,8 @@ class Types::Auth::User < GraphQL::Schema::Object
   field :email, String, null: false do
     description 'Email address of the user'
   end
+
+  def email
+    email.downcase
```
Comment (ferland182) [inline app/graphql/types/auth/user.rb]: The email should be saved with the formatted value so the graphql type will always be downcase. The update can be made here: https://github.com/VetReseau/graphql-auth/blob/eedfa5c190bdf0b3a0f4604c772f38b6867ae36d/app/graphql/mutations/auth/sign_up.rb#L24 and everywhere a user is created into an app that uses this gem

**Review by ferland182 — CHANGES_REQUESTED** [review]

File: app/graphql/types/auth/user.rb (hunk 2 above)
Comment (maxvet) [reply app/graphql/types/auth/user.rb]: @ferland182 after testing it, I realized that the email field is already "downcased". There's nothing to fix actually. I removed this code

**Comment by ferland182** [comment]
@maxvet this code make sure that the email used for **signing the user in** will be downcased, so I think this is what you need. If we want to make sure that the email used for the **sign up** behaves the same, we should make the same for the email field from that form. 

I am aware that we don't use that for the VetLocum api, so I am ok to merge this PR. We can open another PR to enforce this behaviour for the the sign up form for this gem. You can do it, I don't have write access to merge it

**Pull request merged**